docs/DeveloperGuide.md

```diff
@@ -6,33 +6,230 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### The Overall UML Diagram:
 
+![UML Diagram](assets/images/mathGenius.png)    
+
+
+### The overall Sequence Diagram:
+![Sequence Diagram](assets/images/sequenceDiagram.png)    
+
+
+
+### problemGenerator Component
+API: [ProblemGenerator.java](../src/main/java/seedu/duke/ProblemGenerator.java)
+
+the main idea in the design of problemGenerator is that it take 4 parameters and generate a test which include some number of problems
+
+
+when the gengerator is triggered, users enter a command like this:
+
+    generate -t [operators] -n [number] -d [maximum digit] -l[length]
+for example , user can enter:
+
+    generate -t +-*/ -n 10 -d 2 -l 3
+
+in which case the parseCommand() will take the command and pass the parameters to generete().
+
+generate() function will choose 3 random operand with max digits of 2, take a random operation from the operation set of  $+,-,*,/$ to form a single problem ,and loop the execution for 10 times to form a test
+
+if user missed some parameters ,defaultOptions() function will use pre-set default options and invoke UI to print a message for every missing parameter
+
+the answer of the problem is calculated by calculator class while generating a problem, answer is stored in the problem class
+
+### Checker Component  
+**API：[Checker.java](../src/main/java/seedu/duke/Checker.java)**   
+**How the `Checker` work:**  
+1. Every `Checker` was created with a `Test` class.   
+2. The `Checker` will ask user's input and compare the answer up to 2 decimal tolerance.   
+3. If the user input is not a number, the answer will automatically viewed as a incorrect answer.   
+4. The accuracy and the user's answers will be stored for UI or other class to access with the specific function.   
+5. The checker will also store the times that user use to caculate for the problemset.  
+6. The checker will also store the questions that user gave a wrong answer as well as the wrong answer.
+    
+**`Pseudo code` for reference:**  
+```
+# the brief pseudo code for how to check the answer
+
+correct = 0
+isCorrect = []
+wrongProblem = []
+wrongAnswer = []
+for problem in problem set: 
+    UI.PrintProblem
+    answer <- user_input
+    if answer - problem.answer < teleranceRange
+        correct <- correct+1
+        isCorrect.append("1")
+    else
+        isCorrect.append("0")
+        wrongProblem.append(problem)
+        wrongAnswer.append(userInput)
+        continue
+
+```
+**next to be added for `Checker.java`:**  
+1. Support check on more types of problems (i.e Quadratic equation of one variable)   
+2. Give some explanation of the math problems.   
+3. Store the answer in array format to hand the function and matrix answer.
+   
+* To be added
+
+### Record Component - Design
+
+API: [Record.java](../src/main/java/seedu/duke/Record.java)
```

Comment:
API link not working - maybe look into this?
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/111263799/ef6f87c9-c21f-4553-b6dd-5aec31f54137)